### PR TITLE
Fix environment variable read issues and potential bugs

### LINKED ISSUE #976: Feathr version: 0.10.4-rc1 not working getting Value Error in Client step but 0.9.0 worked previously
hey team im getting a Value error after the new version release same code is working with 0.9.0 im trying to use 0.9.0 im getting installation error 
code : ! pip install "git+https://github.com/feathr-ai/feathr.git@0.9.0#subdirectory=feathr_project"
Python interpreter will be restarted.
Collecting git+https://github.com/feathr-ai/feathr.git@0.9.0#subdirectory=feathr_project
  Cloning https://github.com/feathr-ai/feathr.git (to revision 0.9.0) to /tmp/pip-req-build-xtmzuh6h
  Running command git clone -q https://github.com/feathr-ai/feathr.git /tmp/pip-req-build-xtmzuh6h
  WARNING: Did not find branch or tag '0.9.0', assuming revision or ref.
  Running command git checkout -q 0.9.0
  error: pathspec '0.9.0' did not match any file(s) known to git
WARNING: Discarding git+https://github.com/feathr-ai/feathr.git@0.9.0#subdirectory=feathr_project. Command errored out with exit status 1: git checkout -q 0.9.0 Check the logs for full command output.
ERROR: Command errored out with exit status 1: git checkout -q 0.9.0 Check the logs for full command output.
CalledProcessError: Command 'pip --disable-pip-version-check install git+https://github.com/feathr-ai/feathr.git@0.9.0#subdirectory=feathr_project' returned non-zero exit status 1.

#with Latest version :Feathr version: 0.10.4-rc1 

code: client = FeathrClient(
    config_path=config_path,
    credential=credential,
    use_env_vars=False,
)
Error is below: 

2023-01-17 13:29:53.037 | INFO     | feathr.utils._env_config_reader:get:65 - Config secrets__azure_key_vault__name is not found in the environment variable, configuration file, or the remote key value store.
2023-01-17 13:29:53.041 | INFO     | feathr.utils._env_config_reader:get:65 - Config offline_store__s3__s3_enabled is not found in the environment variable, configuration file, or the remote key value store.
2023-01-17 13:29:53.042 | INFO     | feathr.utils._env_config_reader:get:65 - Config offline_store__jdbc__jdbc_enabled is not found in the environment variable, configuration file, or the remote key value store.
2023-01-17 13:29:53.043 | INFO     | feathr.utils._env_config_reader:get:65 - Config offline_store__snowflake__snowflake_enabled is not found in the environment variable, configuration file, or the remote key value store.
2023-01-17 13:29:53.045 | INFO     | feathr.utils._env_config_reader:get:65 - Config spark_config__databricks__feathr_runtime_location is not found in the environment variable, configuration file, or the remote key value store.
ValueError: 
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<command-4053299415941067> in <cell line: 1>()
----> 1 client = FeathrClient(config_path=config_path)

/local_disk0/.ephemeral_nfs/envs/pythonEnv-a646cfc1-2561-408f-a74f-d28a802ad326/lib/python3.9/site-packages/feathr/client.py in __init__(self, config_path, local_workspace_dir, credential, project_registry_tag, use_env_vars)
    183                 )
    184 
--> 185         self._construct_redis_client()
    186 
    187         self.secret_names = []

/local_disk0/.ephemeral_nfs/envs/pythonEnv-a646cfc1-2561-408f-a74f-d28a802ad326/lib/python3.9/site-packages/feathr/client.py in _construct_redis_client(self)
    482             port=port,
    483             password=password,
--> 484             ssl=self._str_to_bool(ssl_enabled))
    485         self.logger.info('Redis connection is successful and completed.')
    486         self.redis_client = redis_client

/local_disk0/.ephemeral_nfs/envs/pythonEnv-a646cfc1-2561-408f-a74f-d28a802ad326/lib/python3.9/site-packages/feathr/client.py in _str_to_bool(self, s)
    468             return False
    469         else:
--> 470             raise ValueError # evil ValueError that doesn't tell you what the wrong value was
    471 
    472     def _construct_redis_client(self):

ValueError:

### PR DESCRIPTION
**Pull request by xiaoyongzhu**
This PR fixes a few issues, mostly:
- relaxing the version check for a few dependencies as we are only using its basic dependency
- Fixing bugs like https://github.com/feathr-ai/feathr/issues/976

### CODE REVIEW

File: feathr_project/setup.py [hunk 1]
```diff
@@ -61,26 +61,28 @@
     include_package_data=True,
     # consider
     install_requires=[
-        "click<=8.1.3",
```
Comment (Yuqing-cat) [inline feathr_project/setup.py:64]: I've noticed several bugs caused by unexpected version conflict before. There are some potential risk removing those version ping before an overall validation.

Comment (xiaoyongzhu) [reply feathr_project/setup.py:64]: I think for those packages that are non-essential, like pandas or click, where we are just using its basic functionalities, we should remove those restrictions to make it easier to install.

For some versions that caused issues or we know won't work, like protobuf, or azure dependencies, we should have version restriction.

**Comment by blrchen** [comment]
I see two changes in this PR

1. Fix 976
2. Left the pip deps version restriction

1 should be merged to main quickly as it's fixing issue on critial path. 2 might require more manual test and can take some time to get it merged. Can I suggest we split into two PRs so 1 can get merged quickly and make more room for 2 to get tested.

**Comment by xiaoyongzhu** [comment]
> I see two changes in this PR
> 
> 1. Fix 976
> 2. Left the pip deps version restriction
> 
> 1 should be merged to main quickly as it's fixing issue on critial path. 2 might require more manual test and can take some time to get it merged. Can I suggest we split into two PRs so 1 can get merged quickly and make more room for 2 to get tested.

Good point, this is reverted.

File: feathr_project/feathr/client.py [hunk 2]
```diff
@@ -459,15 +459,16 @@ def _construct_redis_key(self, feature_table, key):
             key = self._COMPOSITE_KEY_SEPARATOR.join(key)
         return feature_table + self._KEY_SEPARATOR + key
 
-    def _str_to_bool(self, s):
+    def _str_to_bool(self, s, variable_name = None):
         """Define a function to detect convert string to bool, since Redis client sometimes require a bool and sometimes require a str
         """
         if s == 'True' or s == True:
```
Comment (blrchen) [inline feathr_project/feathr/client.py:465]: Suggest support string true and false here. (lowercase). Right now if user defines with 
ssl_enabled: true or ssl_enabled: TRUE, the config might not be picked up correctly.

On a side note, should redis init be decoupled from feathr client init? I am having concern to see redis init happen in a critical path.

**Review by blrchen — APPROVED** [review]
:shipit:

File: feathr_project/feathr/client.py (hunk 2 above)
Comment (xiaoyongzhu) [reply feathr_project/feathr/client.py]: good point

**Review by Yuqing-cat — APPROVED** [review]

**Pull request merged**